src/loaders/jobs/taskfarmer/task_generator.py

```diff
@@ -0,0 +1,247 @@
+import argparse
+import datetime
+import os
+import subprocess
+
+from src.loaders.common import loader_common_names
+
+
+'''
+
+Create required documents and scripts for the TaskFarmer Workflow Manager.
+
+usage: task_generator.py [-h] --tool {checkm2,gtdb_tk} --kbase_collection KBASE_COLLECTION --load_ver LOAD_VER --source_data_dir SOURCE_DATA_DIR
+                         [--root_dir ROOT_DIR] [--submit_job] [--no_submit_job]
+
+optional arguments:
+  -h, --help            show this help message and exit
+
+required named arguments:
+  --tool {checkm2,gtdb_tk}
+                        Name of tool to be executed. (e.g. gtdb_tk, checkm2, etc.)
+  --kbase_collection KBASE_COLLECTION
+                        KBase collection identifier name (e.g. GTDB).
+  --load_ver LOAD_VER   KBase load version (e.g. r207.kbase.1).
+  --source_data_dir SOURCE_DATA_DIR
+                        Source data (genome files) directory. (e.g. /global/cfs/cdirs/kbase/collections/sourcedata/GTDB/r207
+
+optional arguments:
+  --root_dir ROOT_DIR   Root directory for the collections project. (default: /global/cfs/cdirs/kbase/collections)
+  --submit_job          Submit job to slurm
+  --no_submit_job       Do not submit job to slurm
+'''
+
+TOOLS_AVAILABLE = ['checkm2', 'gtdb_tk']
+CHUNK_SIZE = 1000
```

Comment:
Is this reasonable for all tools or should it be customized per tool? I also assume all the (Perlmutter?) nodes have the same specs and this it optimized for those nodes? If so can you add a comment to that effect?

Reply:
👍 updated it so that different tools can have different chunk size.

Reply:
:+1: